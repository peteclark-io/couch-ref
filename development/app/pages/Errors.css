@import "../css/branding.css";

.splash {
   position: absolute;
   top: 0;
   left: 0;
   width: 100%;
   height: 100%;
   background-color: var(--brand-danger);
   text-align: center;
}

.error {
   font-family: var(--font-primary);
   color: var(--brand-secondary);
   font-size: 6em;

   margin-top: 30vh;
}

.message {
   font-family: var(--font-secondary);
   color: var(--brand-secondary);
   margin-top: 20px;
}

.emoji {
  font-size: 0.5em;
}
