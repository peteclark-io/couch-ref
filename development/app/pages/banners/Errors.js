'use strict';

import React from 'react';
import {connect} from 'react-redux';
import styles from './Errors.css';

const Errors = React.createClass({

   propTypes: {
      errors: React.PropTypes.object
   },

   render: function() {
      var codes = Object.keys(this.props.errors);
      var error = undefined;
      if (codes.length > 0){
         error = this.props.errors[codes[0]];
      }

      return (
         <div className={styles.splash}>
            <h1 className={styles.error}>Oh No!</h1>
            <h3 className={styles.message}>Something went wrong! Please refresh the page. <span className={styles.emoji}>ðŸ˜«</span></h3>
            {
               error ?
               <div>
                  <h5 className={styles.message}>For the developers...</h5>
                  <h5 className={styles.message}>{error.code}: {error.message}</h5>
               </div>
               : null
            }
         </div>
      );
   }
});

const getErrors = (state = {errors: {}}) => {
   return state.errors;
};

const mapStateToProps = (state) => {
   return {
      errors: getErrors(state)
   };
};

const LiveErrors = connect(
   mapStateToProps
)(Errors);

export function component(){
   return Errors;
}

export default LiveErrors;
