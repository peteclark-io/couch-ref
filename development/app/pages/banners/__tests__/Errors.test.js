'use strict';

import React from 'react';
import {component} from '../Errors.js';
import renderer from 'react-test-renderer';

test('Errors page won\'t change much.', () => {
   const errors = {
      'auth/made-up-error': {
         code: 'auth/made-up-error',
         message: 'Some made up error.'
      }
   }

   const Errors = component();
   const rendered = renderer.create(
      <Errors errors={errors}/>
   );
   let tree = rendered.toJSON();
   expect(tree).toMatchSnapshot();
});
