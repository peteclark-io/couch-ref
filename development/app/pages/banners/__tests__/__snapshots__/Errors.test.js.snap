exports[`test Errors page won't change much. 1`] = `
<div
  className="splash">
  <h1
    className="error">
    Oh No!
  </h1>
  <h3
    className="message">
    Something went wrong! Please refresh the page. 
    <span
      className="emoji">
      ðŸ˜«
    </span>
  </h3>
  <div>
    <h5
      className="message">
      For the developers...
    </h5>
    <h5
      className="message">
      auth/made-up-error
      : 
      Some made up error.
    </h5>
  </div>
</div>
`;
